File: mobile/src/main/java/com/siliconlabs/bledemo/Browser/Model/Logs/Log.java
package com.siliconlabs.bledemo.Browser.Model.Logs;

import com.siliconlabs.bledemo.Browser.Model.LogType;

import java.text.SimpleDateFormat;
import java.util.Calendar;

public class Log {
    private String logTime;
    private String logInfo;
    private LogType logType;
    private String deviceAddress;

    public String getLogTime() {
        return logTime;
    }

    public void setLogTime(String logTime) {
        this.logTime = logTime;
    }

    public String getLogInfo() {
        return logInfo;
    }

    public void setLogInfo(String logInfo) {
        this.logInfo = logInfo;
    }

    public LogType getLogType() {
        return logType;
    }

    public void setLogType(LogType logType) {
        this.logType = logType;
    }

    public String getDeviceAddress() {
        return deviceAddress;
    }

    public void setDeviceAddress(String deviceAddress) {
        this.deviceAddress = deviceAddress;
    }

    public static String getTime() {
        Calendar calendar = Calendar.getInstance();
        SimpleDateFormat formatter = new SimpleDateFormat("HH:mm:ss.SSS");
        return formatter.format(calendar.getTime());
    }

    public static String getDeviceName(String name) {
        if (name == null) {
            return "N/A";
        }
        return name;
    }

}

File: mobile/src/main/java/com/siliconlabs/bledemo/Browser/Models/Logs/Log.kt
package com.siliconlabs.bledemo.browser.models.logs

import com.siliconlabs.bledemo.browser.models.LogType
import java.text.SimpleDateFormat
import java.util.*

open class Log {
    var logTime: String = ""
    var logInfo: String = ""
    var logType: LogType? = null
    var deviceAddress: String = ""

    companion object {
        fun getTime(): String {
            val calendar = Calendar.getInstance()
            val formatter = SimpleDateFormat("HH:mm:ss.SSS")
            return formatter.format(calendar.time)
        }

        fun getDeviceName(name: String?): String {
            return name ?: "N/A"
        }
    }
}
